app/src/main/java/tk/zedlabs/wallportal/repository/PopularDataSource.kt

```diff
@@ -8,68 +8,79 @@ import kotlinx.coroutines.launch
 import tk.zedlabs.wallportal.data.JsonApi
 import tk.zedlabs.wallportal.data.RetrofitService
 import tk.zedlabs.wallportal.models.WallHavenResponse
-import java.lang.Exception
 
 class PopularDataSource(private val scope: CoroutineScope) : PageKeyedDataSource<Int, WallHavenResponse>() {
 
-    val PAGE_SIZE = 24
-    val FIRST_PAGE = 1
-    val queryParam = "minimal||vaporwave||retrowave||noir"
-    val sorting = "views"
-    var  jsonApi : JsonApi = RetrofitService.createService(JsonApi::class.java)
+    companion object {
+        private const val FIRST_PAGE = 1
+    }
+
+    private val queryParam = "minimal||vaporwave||retrowave||noir"
+    private val sorting = "views"
+    var jsonApi: JsonApi = RetrofitService.createService(JsonApi::class.java)
 
-    override fun loadInitial(params: LoadInitialParams<Int>,callback: LoadInitialCallback<Int, WallHavenResponse>) {
+    override fun loadInitial(
+        params: LoadInitialParams<Int>,
+        callback: LoadInitialCallback<Int, WallHavenResponse>
+    ) {
 
         scope.launch {
             try {
                 val response = jsonApi.getImageList(queryParam, sorting, FIRST_PAGE)
-                when{
+                when {
                     response.isSuccessful -> {
-                        callback.onResult(response.body()!!.data!!,null,FIRST_PAGE+1)
+                        callback.onResult(response.body()?.data ?: emptyList(), null, FIRST_PAGE + 1)
```

Comment:
why should we use `?: emptyList()` instead of !!, I heard it was better practice to use `!!`, also more concise.

Reply:
to avoid nullPointerException if returned null. !! operator lead to NPE